### PR TITLE
Ensure Cart block supports the WCBlock level global styles for `Blocks » Product Title`

### LINKED ISSUE #8829: Ensure Cart block supports the WCBlock-level global styles for `Blocks » Product Title » Typography`
Currently, the Cart block partially supports the default WCBlock-level global styles for `Blocks » Product Title » layout`. This issue aims to ensure that the Cart block fully supports the default WCBlock-level global styles.

## Steps to reproduce

1. Go to `WP Admin » Apperance » Editor`.
2. Open the `Styles` sidebar.
3. Open the `Blocks » Product Title » Typography` settings.
4. Adjust the settings.
5. Notice it is getting reflected properly on the frontend and editor. 

## Screenshots
| Settings | Editor | Frontend |
|---|---|---|
|image|image|image|

### LINKED ISSUE #8830: Ensure Cart block supports the WCBlock-level global styles for `Blocks » Product Title » Layout`
Currently, the Cart block partially supports the default WCBlock-level global styles for `Blocks » Product Title » Layout`. This issue aims to ensure that the Cart block fully supports the default WCBlock-level global styles.

## Steps to reproduce

1. Go to `WP Admin » Apperance » Editor`.
2. Open the `Styles` sidebar.
3. Open the `Blocks » Product Title » Layout` settings.
4. Adjust the settings.
5. Notice it is getting reflected properly on the front end and editor. 

## Screenshots
| Settings | Editor | Frontend |
|---|---|---|
|image|image|image|

### LINKED ISSUE #8831: Ensure Cart block supports the WCBlock-level global styles for `Blocks » Product Title » Colors`
Currently, the Cart block partially supports the default WCBlock-level global styles for `Blocks » Product Title » Colors`. This issue aims to ensure that the Cart block fully supports the WCBlock theme-level global styles.

## Steps to reproduce

1. Go to `WP Admin » Apperance » Editor`.
2. Open the `Styles` sidebar.
3. Open the `Blocks » Product Title » Colors` settings.
4. Adjust the settings.
5. Notice it is getting reflected properly on the front end and editor. 

## Screenshots
| Settings | Editor | Frontend |
|---|---|---|
|image|image|image|

### PR DESCRIPTION
**Pull request by tarunvijwani**
Fixes #8829
Fixes #8830
Fixes #8831 

Currently, the Cart block partially supports the WCBlock level global styles for Blocks » Product Title. 
This PR ensures the Cart block fully supports the WCBlock level global styles.

### Screenshots

> **Note**
> The following screenshots, show the results of this PR. Please note that the product titles do not inherit the margin settings from `Blocks » Product Title » Layout` to prevent the layout from breaking.

#### Blocks » Product Title » Typography

Screen Shot 2023-04-05 at 15 48 03

#### Blocks » Product Title » Color

Screen Shot 2023-04-05 at 15 49 19

#### Blocks » Product Title » Layout

Screen Shot 2023-04-05 at 15 50 59

#### Automated Tests
* [ ] Changes in this PR are covered by Automated Tests.
  * [ ] Unit tests
  * [ ] E2E tests

#### User Facing Testing

1. Install and activate the [Beaumont](https://wordpress.org/themes/beaumont/) theme.
2. Create a test page with the Checkout block.
3. Go to `WP Admin » Settings » Reading`.
4. Define the test page from step 2. as a static page.
Screenshot 2023-03-29 at 11 15 29

5. Go to `WP Admin » Apperance » Editor`.
6. Open the `Styles` sidebar.
8. Open the `Blocks` section.
9. Search for the `Product Title` block.
Screen Shot 2023-04-05 at 15 27 51

10. Open the `Blocks » Product Title » Typography` settings.
11. Adjust `FONT`, `SIZE`, `FONT WEIGHT`, `LINE HEIGHT` and `LETTER SPACE`.
12. Open the test page from step 2. both in the post editor and the frontend.
13. Verify that the product titles show the typography settings as defined in step 10.
14. Open the `Blocks » Product Title » Colors` settings.
15. Adjust `Background` and `Text` colors. (Note that `Headings` color is not used in this block.)
16. Open the test page from step 2. both in the post editor and the frontend.
17. Verify that the product titles show the colors as defined in step 14.
18. Open the `Blocks » Product Title » Layout` settings.
19. Adjust the `MARGIN`.
20. Verify that the product titles do not inherit the margin settings from step 18.

**Note: Cart product title styles should match the product titles in the products block so that product titles look consistent throughout the site.** 

* [ ] Do not include in the Testing Notes 

### WooCommerce Visibility

* [x] WooCommerce Core
* [ ] Feature plugin
* [ ] Experimental

### Changelog

> Ensure Cart block supports the WCBlock level global styles for `Blocks » Product Title`

### CODE REVIEW

**Review by nielslange — COMMENTED** [review]
Thanks for working on this, @tarunvijwani. I reviewed the PR with the [Beaumont](https://andersnoren.se/teman/beaumont-wordpress-theme/) and [Twenty Twenty-Three](https://wordpress.org/themes/twentytwentythree/) themes, getting the following results:

### Support
- 🟢 does work
- 🟠 cannot be tested
- 🔴 does not work

#### Post Editor
- 🟢 Blocks » Product Title » Typography » FONT
- 🔴 Blocks » Product Title » Typography »  SIZE
- 🟢 Blocks » Product Title » Typography » FONT WEIGHT
- 🟢 Blocks » Product Title » Typography » LINE HEIGHT
- 🟢 Blocks » Product Title » Color » Background
- 🟢 Blocks » Product Title » Color » Text
- 🟠 Blocks » Product Title » Color » Headings
- 🟢 Blocks » Product Title » Layout

#### Frontend
- 🟢 Blocks » Product Title » Typography » FONT
- 🔴 Blocks » Product Title » Typography »  SIZE
- 🟢 Blocks » Product Title » Typography » FONT WEIGHT
- 🟢 Blocks » Product Title » Typography » LINE HEIGHT
- 🟢 Blocks » Product Title » Color » Background
- 🔴 Blocks » Product Title » Color » Text
- 🟠 Blocks » Product Title » Color » Headings
- 🟢 Blocks » Product Title » Layout

I also noticed that only the product titles of the Cross-Sells products are affected, but not the product titles within the Cart Line Items section.

Based on the screenshots in the PR, it appears that the product titles of both sections should be impacted.

Screen Shot 2023-04-04 at 12 30 32

To make it easier for reviewers, I wonder if we should list all options in the test steps. An example of what I mean, can be seen on https://github.com/woocommerce/woocommerce-blocks/pull/8918:

Screenshot 2023-04-04 at 12 55 09

File: assets/js/base/components/cart-checkout/cart-line-items-table/cart-line-item-row.tsx [hunk 1]
```diff
@@ -237,13 +237,16 @@ const CartLineItemRow: React.ForwardRefExoticComponent<
...
 				<td className="wc-block-cart-item__product">
 					<div className="wc-block-cart-item__wrap">
-						<ProductName
-							disabled={
-								isPendingDelete || isProductHiddenFromCatalog
-							}
-							name={ name }
-							permalink={ permalink }
-						/>
+						<span className="wc-block-components-product-title">
```
Comment (mikejolley) [inline assets/js/base/components/cart-checkout/cart-line-items-table/cart-line-item-row.tsx]: I believe wrapping this with a classname is the wrong approach. It looks as though `wc-block-components-product-title` is used within the product title block and is not exposed as a component. **Why don't we expose a component instead?** if the styles and component are to be shared, that makes the most sense.

- Create a base component called product title
- Move the code from the product title block into the component. Note that it accepts `tag` so you can define the h* tag or pass a span.
- Make both the product title block and the cart item list use the new component.

**Review by mikejolley — COMMENTED** [review]
Works but we should revise how the components are used rather than insert a span using another components classname.

**Review by Aljullu — COMMENTED** [review]
I tested and it works well, but I have to say I'm a bit unsure if we should go ahead with this. Let me explain:

The Product Title global styles should be applied to all Product Title blocks, but nothing else IMO. In the case of the Cart block, those product titles are not the Product Title block. Instead, they are a component displaying the product title. I'm wary of applying the styles from a block to a different component, as that might cause future issues and be confusing for users. For example, it doesn't allow overriding the global styles: ie, imagine I want product titles to be bold _except_ when displayed in the Cart block, I wouldn't be able to do that with this approach.

Something similar happened with the Sale Badge. We have the Sale Badge block and the sale badge component rendered inside the Product Image block. Currently, styles applied to the Sale Badge block are applied to the sale badge component. But IMO that's confusing and prone to break in the future.

I posted a question regarding this in https://github.com/woocommerce/woocommerce-blocks/issues/8151.

File: assets/js/base/components/cart-checkout/cart-line-items-table/cart-line-item-row.tsx (hunk 1 above)
Comment (nielslange) [reply assets/js/base/components/cart-checkout/cart-line-items-table/cart-line-item-row.tsx]: > I believe wrapping this with a classname is the wrong approach. It looks as though `wc-block-components-product-title` is used within the product title block and is not exposed as a component. **Why don't we expose a component instead?** if the styles and component are to be shared, that makes the most sense.
> 
> * Create a base component called product title
> * Move the code from the product title block into the component. Note that it accepts `tag` so you can define the h* tag or pass a span.
> * Make both the product title block and the cart item list use the new component.

Thanks for your review, @mikejolley. Creating the base component sounds like a good approach. I'll do that now.

**Comment by nielslange** [comment]
Thanks for your review, @Aljullu. In https://github.com/woocommerce/woocommerce-blocks/pull/8939#pullrequestreview-1374589000, @mikejolley suggested creating a base component called product title and using that instead of applying styles from a different block. I'll keep working on this approach now.

File: assets/js/settings/blocks/constants.ts [hunk 2]
```diff
@@ -30,15 +30,15 @@ export const WC_BLOCKS_IMAGE_URL = blocksConfig.pluginUrl + 'images/';
...
-export const LOGIN_URL = STORE_PAGES.myaccount.permalink
+export const CHECKOUT_PAGE_ID = STORE_PAGES.checkout?.id;
+export const CHECKOUT_URL = STORE_PAGES.checkout?.permalink;
+export const PRIVACY_URL = STORE_PAGES.privacy?.permalink;
+export const PRIVACY_PAGE_NAME = STORE_PAGES.privacy?.title;
+export const TERMS_URL = STORE_PAGES.terms?.permalink;
+export const TERMS_PAGE_NAME = STORE_PAGES.terms?.title;
+export const CART_PAGE_ID = STORE_PAGES.cart?.id;
+export const CART_URL = STORE_PAGES.cart?.permalink;
+export const LOGIN_URL = STORE_PAGES.myaccount?.permalink
```
Comment (nielslange) [inline assets/js/settings/blocks/constants.ts:41]: These changes ensure that the preview does not break in Storybook.

**Comment by tarunvijwani** [comment]
We're discussing the final approach on slack: p1681203933835379-slack-C02UBB1EPEF

**Comment by tarunvijwani** [comment]
Closing the PR as per the discussion on Slack: p1681728763577949/1681203933.835379-slack-C02UBB1EPEF

The Product Title block and Product Title component in the Cart Line Items block are different; hence they should have different global style options. The Product Title component should inherit the parent's Global styles, i.e., Cart Line Items inner block instead of the Product Title block.

**Pull request closed without merging**